game-core/src/main/java/games/strategy/debug/error/reporting/ErrorReportConfiguration.java

```diff
@@ -32,12 +32,9 @@
         .map(ErrorReportUploadAction::new);
   }
 
-  private static Optional<BiConsumer<JFrame, UserErrorReport>> newFromRemoteProperties() {
-    return LobbyPropertyFetcherConfiguration.lobbyServerPropertiesFetcher()
-        .fetchLobbyServerProperties()
-        .map(
-            props ->
-                new ErrorReportUploadAction(
-                    ErrorReportClientFactory.newErrorUploader(props.getHttpServerUri())));
+  private static BiConsumer<JFrame, UserErrorReport> newFromRemoteProperties() {
+    final LobbyServerProperties props =
+        LobbyPropertyFetcherConfiguration.lobbyServerPropertiesFetcher().fetchLobbyServerProperties();
+    return new ErrorReportUploadAction(ErrorReportClientFactory.newErrorUploader(props.getHttpServerUri()));
```

Comment:
I think there may be a latent bug here.  Because `OFFLINE_STRATEGY` never was hit, there's a possibility that `props.getHttpServerUri()` is going to return `null` here.  In fact, I'm pretty sure it will if anything other than the `LobbyServerProperties` returned from `LobbyServerPropertiesFetcher#getRemoteProperties()` below is used.  The other two branches never set the `httpServerUri` field.

It might be clearer to make the `@Nullable` getters in `LobbyServerProperties` return an `Optional` instead.  That way the compiler will get in your face to alert you to the possibility of a missing value.

Reply:
> I think there may be a latent bug here. Because OFFLINE_STRATEGY never was hit

The unused code was a correct indicator of a problem. The 'lastUsed' property had an incorrect `getOrThrow` when it should have allowed an empty return value. That then fixes the unnecessary Optional. A fix for that is here: https://github.com/triplea-game/triplea/pull/4521

> It might be clearer to make the @Nullable getters in LobbyServerProperties return an Optional instead.

I've been resisting to update that API in that manner, and I find it interesting that it's a much clearer contract rather than an empty String. While updating the API/implementation of `LobbyServerProperties`, noting the `Integer` -> `int`, I think we want that to be a `@Nonnull Integer` to force the value to be set during construction. Otherwise we could leave it off and the value would default to zero. 

The API update and builder requirement update are both in: https://github.com/triplea-game/triplea/pull/4522

Reply:
> While updating the API/implementation of `LobbyServerProperties`, noting the `Integer` -> `int`, I think we want that to be a `@Nonnull Integer` to force the value to be set during construction. Otherwise we could leave it off and the value would default to zero.

Ah, yes, I missed that.  I had assumed that since Lombok provided the `@Builder.Default` annotation, it assumed all fields were required by default.  But I guess there's really no magical way to do that for primitives without using boxed (thus nullable) types.

I guess the way I would have done that by hand is use an `Integer` field in the builder but an `int` field in the target and enforce the non-nullness of the `Integer` in the `build()` method.  But such are the sacrifices for using auto-generated code, I guess. :smile:

There's apparently been an ongoing discussion about required fields in a Lombok builder that it recently generated an official [feature proposal](https://github.com/rzwitserloot/lombok/wiki/FEATURE-IDEA:-%22Mandatory%22-fields-with-@Builder).